commit 0d5de88c4b5ad369fa41953a9831db3f27868b09
Author: Piotr Osiewicz <piotr@zed.dev>
Date:   Thu Jul 25 22:48:42 2024 +0200

    chore: Bump Rust version to 1.80 (#15186)
    
    Release Notes:
    
    - N/A

diff --git b/Dockerfile a/Dockerfile
index 7d4b6b7..6c8d3da 100644
--- b/Dockerfile
+++ a/Dockerfile
@@ -1,6 +1,6 @@
 # syntax = docker/dockerfile:1.2
 
-FROM rust:1.80-bookworm as builder
+FROM rust:1.79-bookworm as builder
 WORKDIR app
 COPY . .
 
diff --git b/crates/assistant/src/slash_command/diagnostics_command.rs a/crates/assistant/src/slash_command/diagnostics_command.rs
index 1e9d050..723deb1 100644
--- b/crates/assistant/src/slash_command/diagnostics_command.rs
+++ a/crates/assistant/src/slash_command/diagnostics_command.rs
@@ -33,7 +33,7 @@ impl DiagnosticsSlashCommand {
         if query.is_empty() {
             let workspace = workspace.read(cx);
             let entries = workspace.recent_navigation_history(Some(10), cx);
-            let path_prefix: Arc<str> = Arc::default();
+            let path_prefix: Arc<str> = "".into();
             Task::ready(
                 entries
                     .into_iter()
diff --git b/crates/assistant/src/slash_command/file_command.rs a/crates/assistant/src/slash_command/file_command.rs
index c70c44f..849ece1 100644
--- b/crates/assistant/src/slash_command/file_command.rs
+++ a/crates/assistant/src/slash_command/file_command.rs
@@ -29,7 +29,7 @@ impl FileSlashCommand {
             let workspace = workspace.read(cx);
             let project = workspace.project().read(cx);
             let entries = workspace.recent_navigation_history(Some(10), cx);
-            let path_prefix: Arc<str> = Arc::default();
+            let path_prefix: Arc<str> = "".into();
             Task::ready(
                 entries
                     .into_iter()
diff --git b/crates/editor/src/editor.rs a/crates/editor/src/editor.rs
index cd63548..f402be2 100644
--- b/crates/editor/src/editor.rs
+++ a/crates/editor/src/editor.rs
@@ -408,7 +408,6 @@ impl EditorActionId {
 type BackgroundHighlight = (fn(&ThemeColors) -> Hsla, Arc<[Range<Anchor>]>);
 type GutterHighlight = (fn(&AppContext) -> Hsla, Arc<[Range<Anchor>]>);
 
-#[derive(Default)]
 struct ScrollbarMarkerState {
     scrollbar_size: Size<Pixels>,
     dirty: bool,
@@ -422,6 +421,17 @@ impl ScrollbarMarkerState {
     }
 }
 
+impl Default for ScrollbarMarkerState {
+    fn default() -> Self {
+        Self {
+            scrollbar_size: Size::default(),
+            dirty: false,
+            markers: Arc::from([]),
+            pending_refresh: None,
+        }
+    }
+}
+
 #[derive(Clone, Debug)]
 struct RunnableTasks {
     templates: Vec<(TaskSourceKind, TaskTemplate)>,
@@ -5720,7 +5730,7 @@ impl Editor {
 
         self.transact(cx, |this, cx| {
             this.buffer.update(cx, |buffer, cx| {
-                let empty_str: Arc<str> = Arc::default();
+                let empty_str: Arc<str> = "".into();
                 buffer.edit(
                     deletion_ranges
                         .into_iter()
@@ -5786,7 +5796,7 @@ impl Editor {
 
         self.transact(cx, |this, cx| {
             let buffer = this.buffer.update(cx, |buffer, cx| {
-                let empty_str: Arc<str> = Arc::default();
+                let empty_str: Arc<str> = "".into();
                 buffer.edit(
                     edit_ranges
                         .into_iter()
@@ -8087,7 +8097,7 @@ impl Editor {
             let mut selection_edit_ranges = Vec::new();
             let mut last_toggled_row = None;
             let snapshot = this.buffer.read(cx).read(cx);
-            let empty_str: Arc<str> = Arc::default();
+            let empty_str: Arc<str> = "".into();
             let mut suffixes_inserted = Vec::new();
 
             fn comment_prefix_range(
diff --git b/crates/editor/src/selections_collection.rs a/crates/editor/src/selections_collection.rs
index 7ce3a93..4161792 100644
--- b/crates/editor/src/selections_collection.rs
+++ a/crates/editor/src/selections_collection.rs
@@ -44,7 +44,7 @@ impl SelectionsCollection {
             buffer,
             next_selection_id: 1,
             line_mode: false,
-            disjoint: Arc::default(),
+            disjoint: Arc::from([]),
             pending: Some(PendingSelection {
                 selection: Selection {
                     id: 0,
@@ -398,7 +398,7 @@ impl<'a> MutableSelectionsCollection<'a> {
     }
 
     pub fn clear_disjoint(&mut self) {
-        self.collection.disjoint = Arc::default();
+        self.collection.disjoint = Arc::from([]);
     }
 
     pub fn delete(&mut self, selection_id: usize) {
diff --git b/crates/editor/src/test/editor_test_context.rs a/crates/editor/src/test/editor_test_context.rs
index 89a47ea..302d012 100644
--- b/crates/editor/src/test/editor_test_context.rs
+++ a/crates/editor/src/test/editor_test_context.rs
@@ -327,7 +327,7 @@ impl EditorTestContext {
                 .background_highlights
                 .get(&TypeId::of::<Tag>())
                 .map(|h| h.1.clone())
-                .unwrap_or_else(|| Arc::default())
+                .unwrap_or_else(|| Arc::from([]))
                 .into_iter()
                 .map(|range| range.to_offset(&snapshot.buffer_snapshot))
                 .collect()
diff --git b/crates/file_finder/src/file_finder.rs a/crates/file_finder/src/file_finder.rs
index 9f7b163..74337c1 100644
--- b/crates/file_finder/src/file_finder.rs
+++ a/crates/file_finder/src/file_finder.rs
@@ -998,7 +998,7 @@ mod tests {
                 positions: Vec::new(),
                 worktree_id: 0,
                 path: Arc::from(Path::new("b0.5")),
-                path_prefix: Arc::default(),
+                path_prefix: Arc::from(""),
                 distance_to_relative_ancestor: 0,
             }),
             ProjectPanelOrdMatch(PathMatch {
@@ -1006,7 +1006,7 @@ mod tests {
                 positions: Vec::new(),
                 worktree_id: 0,
                 path: Arc::from(Path::new("c1.0")),
-                path_prefix: Arc::default(),
+                path_prefix: Arc::from(""),
                 distance_to_relative_ancestor: 0,
             }),
             ProjectPanelOrdMatch(PathMatch {
@@ -1014,7 +1014,7 @@ mod tests {
                 positions: Vec::new(),
                 worktree_id: 0,
                 path: Arc::from(Path::new("a1.0")),
-                path_prefix: Arc::default(),
+                path_prefix: Arc::from(""),
                 distance_to_relative_ancestor: 0,
             }),
             ProjectPanelOrdMatch(PathMatch {
@@ -1022,7 +1022,7 @@ mod tests {
                 positions: Vec::new(),
                 worktree_id: 0,
                 path: Arc::from(Path::new("a0.5")),
-                path_prefix: Arc::default(),
+                path_prefix: Arc::from(""),
                 distance_to_relative_ancestor: 0,
             }),
             ProjectPanelOrdMatch(PathMatch {
@@ -1030,7 +1030,7 @@ mod tests {
                 positions: Vec::new(),
                 worktree_id: 0,
                 path: Arc::from(Path::new("b1.0")),
-                path_prefix: Arc::default(),
+                path_prefix: Arc::from(""),
                 distance_to_relative_ancestor: 0,
             }),
         ];
@@ -1044,7 +1044,7 @@ mod tests {
                     positions: Vec::new(),
                     worktree_id: 0,
                     path: Arc::from(Path::new("a1.0")),
-                    path_prefix: Arc::default(),
+                    path_prefix: Arc::from(""),
                     distance_to_relative_ancestor: 0,
                 }),
                 ProjectPanelOrdMatch(PathMatch {
@@ -1052,7 +1052,7 @@ mod tests {
                     positions: Vec::new(),
                     worktree_id: 0,
                     path: Arc::from(Path::new("b1.0")),
-                    path_prefix: Arc::default(),
+                    path_prefix: Arc::from(""),
                     distance_to_relative_ancestor: 0,
                 }),
                 ProjectPanelOrdMatch(PathMatch {
@@ -1060,7 +1060,7 @@ mod tests {
                     positions: Vec::new(),
                     worktree_id: 0,
                     path: Arc::from(Path::new("c1.0")),
-                    path_prefix: Arc::default(),
+                    path_prefix: Arc::from(""),
                     distance_to_relative_ancestor: 0,
                 }),
                 ProjectPanelOrdMatch(PathMatch {
@@ -1068,7 +1068,7 @@ mod tests {
                     positions: Vec::new(),
                     worktree_id: 0,
                     path: Arc::from(Path::new("a0.5")),
-                    path_prefix: Arc::default(),
+                    path_prefix: Arc::from(""),
                     distance_to_relative_ancestor: 0,
                 }),
                 ProjectPanelOrdMatch(PathMatch {
@@ -1076,7 +1076,7 @@ mod tests {
                     positions: Vec::new(),
                     worktree_id: 0,
                     path: Arc::from(Path::new("b0.5")),
-                    path_prefix: Arc::default(),
+                    path_prefix: Arc::from(""),
                     distance_to_relative_ancestor: 0,
                 }),
             ]
diff --git b/crates/fuzzy/src/matcher.rs a/crates/fuzzy/src/matcher.rs
index d3b0208..9b0d628 100644
--- b/crates/fuzzy/src/matcher.rs
+++ a/crates/fuzzy/src/matcher.rs
@@ -404,12 +404,7 @@ mod tests {
 
     #[test]
     fn test_match_multibyte_path_entries() {
-        let paths = vec![
-            "aαbβ/cγdδ",
-            "αβγδ/bcde",
-            "c1️⃣2️⃣3️⃣/d4️⃣5️⃣6️⃣/e7️⃣8️⃣9️⃣/f",
-            "/d/🆒/h",
-        ];
+        let paths = vec!["aαbβ/cγdδ", "αβγδ/bcde", "c1️⃣2️⃣3️⃣/d4️⃣5️⃣6️⃣/e7️⃣8️⃣9️⃣/f", "/d/🆒/h"];
         assert_eq!("1️⃣".len(), 7);
         assert_eq!(
             match_single_path_query("bcd", false, &paths),
diff --git b/crates/fuzzy/src/paths.rs a/crates/fuzzy/src/paths.rs
index 7319251..25927f1 100644
--- b/crates/fuzzy/src/paths.rs
+++ a/crates/fuzzy/src/paths.rs
@@ -120,7 +120,7 @@ pub fn match_fixed_path_set(
             worktree_id,
             positions: Vec::new(),
             path: Arc::from(candidate.path),
-            path_prefix: Arc::default(),
+            path_prefix: Arc::from(""),
             distance_to_relative_ancestor: usize::MAX,
         },
     );
diff --git b/crates/gpui/src/platform/linux/wayland/client.rs a/crates/gpui/src/platform/linux/wayland/client.rs
index 7462819..c97fe24 100644
--- b/crates/gpui/src/platform/linux/wayland/client.rs
+++ a/crates/gpui/src/platform/linux/wayland/client.rs
@@ -395,7 +395,6 @@ impl WaylandClient {
         let qh = event_queue.handle();
 
         let mut seat: Option<wl_seat::WlSeat> = None;
-        #[allow(clippy::mutable_key_type)]
         let mut in_progress_outputs = HashMap::default();
         globals.contents().with_list(|list| {
             for global in list {
@@ -875,7 +874,6 @@ impl Dispatch<wl_surface::WlSurface, ()> for WaylandClientStatePtr {
         let Some(window) = get_window(&mut state, &surface.id()) else {
             return;
         };
-        #[allow(clippy::mutable_key_type)]
         let outputs = state.outputs.clone();
         drop(state);
 
diff --git b/crates/gpui/src/platform/linux/wayland/window.rs a/crates/gpui/src/platform/linux/wayland/window.rs
index 9ec5e9e..c43ec18 100644
--- b/crates/gpui/src/platform/linux/wayland/window.rs
+++ a/crates/gpui/src/platform/linux/wayland/window.rs
@@ -545,7 +545,6 @@ impl WaylandWindowStatePtr {
         }
     }
 
-    #[allow(clippy::mutable_key_type)]
     pub fn handle_surface_event(
         &self,
         event: wl_surface::Event,
diff --git b/crates/gpui/src/shared_string.rs a/crates/gpui/src/shared_string.rs
index efdf323..1aa1bca 100644
--- b/crates/gpui/src/shared_string.rs
+++ a/crates/gpui/src/shared_string.rs
@@ -10,7 +10,7 @@ pub struct SharedString(ArcCow<'static, str>);
 
 impl Default for SharedString {
     fn default() -> Self {
-        Self(ArcCow::Owned(Arc::default()))
+        Self(ArcCow::Owned("".into()))
     }
 }
 
diff --git b/crates/language/src/buffer.rs a/crates/language/src/buffer.rs
index 27e6222..0d2ed42 100644
--- b/crates/language/src/buffer.rs
+++ a/crates/language/src/buffer.rs
@@ -1402,7 +1402,7 @@ impl Buffer {
                 LineEnding::normalize(&mut new_text);
 
                 let diff = TextDiff::from_chars(old_text.as_str(), new_text.as_str());
-                let empty: Arc<str> = Arc::default();
+                let empty: Arc<str> = "".into();
 
                 let mut edits = Vec::new();
                 let mut old_offset = 0;
@@ -1720,7 +1720,7 @@ impl Buffer {
             .get(&self.text.replica_id())
             .map_or(true, |set| !set.selections.is_empty())
         {
-            self.set_active_selections(Arc::default(), false, Default::default(), cx);
+            self.set_active_selections(Arc::from([]), false, Default::default(), cx);
         }
     }
 
diff --git b/crates/language/src/language.rs a/crates/language/src/language.rs
index 03f1432..c07090f 100644
--- b/crates/language/src/language.rs
+++ a/crates/language/src/language.rs
@@ -682,7 +682,7 @@ impl<T> Override<T> {
 impl Default for LanguageConfig {
     fn default() -> Self {
         Self {
-            name: Arc::default(),
+            name: "".into(),
             code_fence_block_name: None,
             grammar: None,
             matcher: LanguageMatcher::default(),
diff --git b/crates/lsp/src/lsp.rs a/crates/lsp/src/lsp.rs
index 30feffa..d3cce5d 100644
--- b/crates/lsp/src/lsp.rs
+++ a/crates/lsp/src/lsp.rs
@@ -389,7 +389,7 @@ impl LanguageServer {
             notification_handlers,
             response_handlers,
             io_handlers,
-            name: Arc::default(),
+            name: "".into(),
             capabilities: Default::default(),
             code_action_kinds,
             next_id: Default::default(),
diff --git b/crates/markdown/src/markdown.rs a/crates/markdown/src/markdown.rs
index 379df9e..915c3b0 100644
--- b/crates/markdown/src/markdown.rs
+++ a/crates/markdown/src/markdown.rs
@@ -216,7 +216,7 @@ impl Selection {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 pub struct ParsedMarkdown {
     source: SharedString,
     events: Arc<[(Range<usize>, MarkdownEvent)]>,
@@ -232,6 +232,15 @@ impl ParsedMarkdown {
     }
 }
 
+impl Default for ParsedMarkdown {
+    fn default() -> Self {
+        Self {
+            source: SharedString::default(),
+            events: Arc::from([]),
+        }
+    }
+}
+
 pub struct MarkdownElement {
     markdown: View<Markdown>,
     style: MarkdownStyle,
diff --git b/crates/multi_buffer/src/multi_buffer.rs a/crates/multi_buffer/src/multi_buffer.rs
index 1478e25..0ee107d 100644
--- b/crates/multi_buffer/src/multi_buffer.rs
+++ a/crates/multi_buffer/src/multi_buffer.rs
@@ -673,7 +673,7 @@ impl MultiBuffer {
                         let mut insertions = Vec::new();
                         let mut original_indent_columns = Vec::new();
                         let mut deletions = Vec::new();
-                        let empty_str: Arc<str> = Arc::default();
+                        let empty_str: Arc<str> = "".into();
                         while let Some(BufferEdit {
                             mut range,
                             new_text,
diff --git b/crates/project/src/project.rs a/crates/project/src/project.rs
index 7bd6378..c937362 100644
--- b/crates/project/src/project.rs
+++ a/crates/project/src/project.rs
@@ -10844,7 +10844,7 @@ impl<'a> fuzzy::PathMatchCandidateSet<'a> for PathMatchCandidateSet {
         } else if self.include_root_name {
             format!("{}/", self.snapshot.root_name()).into()
         } else {
-            Arc::default()
+            "".into()
         }
     }
 
diff --git b/crates/rich_text/src/rich_text.rs a/crates/rich_text/src/rich_text.rs
index 2c4b2ca..16c4473 100644
--- b/crates/rich_text/src/rich_text.rs
+++ a/crates/rich_text/src/rich_text.rs
@@ -31,7 +31,7 @@ impl From<HighlightId> for Highlight {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 pub struct RichText {
     pub text: SharedString,
     pub highlights: Vec<(Range<usize>, Highlight)>,
@@ -43,6 +43,19 @@ pub struct RichText {
         Option<Arc<dyn Fn(usize, Range<usize>, &mut WindowContext) -> Option<AnyView>>>,
 }
 
+impl Default for RichText {
+    fn default() -> Self {
+        Self {
+            text: SharedString::default(),
+            highlights: Vec::new(),
+            link_ranges: Vec::new(),
+            link_urls: Arc::from([]),
+            custom_ranges: Vec::new(),
+            custom_ranges_tooltip_fn: None,
+        }
+    }
+}
+
 /// Allows one to specify extra links to the rendered markdown, which can be used
 /// for e.g. mentions.
 #[derive(Debug)]
diff --git b/crates/tab_switcher/src/tab_switcher.rs a/crates/tab_switcher/src/tab_switcher.rs
index bfbd1aa..232c0b3 100644
--- b/crates/tab_switcher/src/tab_switcher.rs
+++ a/crates/tab_switcher/src/tab_switcher.rs
@@ -264,7 +264,7 @@ impl PickerDelegate for TabSwitcherDelegate {
     type ListItem = ListItem;
 
     fn placeholder_text(&self, _cx: &mut WindowContext) -> Arc<str> {
-        Arc::default()
+        "".into()
     }
 
     fn no_matches_text(&self, _cx: &mut WindowContext) -> SharedString {
diff --git b/crates/worktree/src/worktree.rs a/crates/worktree/src/worktree.rs
index 19c91d9..6834e10 100644
--- b/crates/worktree/src/worktree.rs
+++ a/crates/worktree/src/worktree.rs
@@ -510,7 +510,7 @@ impl Worktree {
                                 }
                             }
                         };
-                        cx.emit(Event::UpdatedEntries(Arc::default()));
+                        cx.emit(Event::UpdatedEntries(Arc::from([])));
                         cx.notify();
                         while let Some((scan_id, _)) = this.snapshot_subscriptions.front() {
                             if this.observed_snapshot(*scan_id) {
@@ -1698,7 +1698,7 @@ impl LocalWorktree {
         let (snapshots_tx, mut snapshots_rx) =
             mpsc::unbounded::<(LocalSnapshot, UpdatedEntriesSet, UpdatedGitRepositoriesSet)>();
         snapshots_tx
-            .unbounded_send((self.snapshot(), Arc::default(), Arc::default()))
+            .unbounded_send((self.snapshot(), Arc::from([]), Arc::from([])))
             .ok();
 
         let worktree_id = cx.entity_id().as_u64();
diff --git b/rust-toolchain.toml a/rust-toolchain.toml
index 9c975c4..8d422bd 100644
--- b/rust-toolchain.toml
+++ a/rust-toolchain.toml
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "1.80"
+channel = "1.79"
 profile = "minimal"
 components = [ "rustfmt", "clippy" ]
 targets = [ "x86_64-apple-darwin", "aarch64-apple-darwin", "x86_64-unknown-linux-gnu", "wasm32-wasi" ]
