--- a/src/core-impl/storage/sql/mysql-shared/MySqlStorage.h
+++ /bsrc/core-impl/storage/sql/mysql-shared/MySqlStorage.h
@@ -31,2 +31,1 @@
-struct st_mysql;
-typedef struct st_mysql MYSQL;
+typedef struct MYSQL MYSQL;

--- a/src/core-impl/storage/sql/mysqlserverstorage/MySqlServerStorage.cpp
+++ b/src/core-impl/storage/sql/mysqlserverstorage/MySqlServerStorage.cpp
@@ -69,1 +69,1 @@
-    my_bool reconnect = true;
+    bool reconnect = true;
