--- mysql-workbench-community-6.3.4-src/library/forms/gtk/src/lf_popover.cpp	2015-06-08 08:30:54.000000000 -0300
+++ /tmp/lf_popover.cpp	2017-01-01 20:20:55.552767268 -0200
@@ -380,7 +380,7 @@
   if (_style == mforms::PopoverStyleTooltip)
   {
     Glib::RefPtr<Gdk::Window> wnd = this->get_window();
-    if (wnd != 0)
+    if (wnd)
     {
       int xx;
       int yy;
@@ -396,7 +396,7 @@
   {
     Gdk::ModifierType mask;
     Glib::RefPtr<Gdk::Display> dsp = Gdk::Display::get_default();
-    if (dsp != 0)
+    if (dsp)
       dsp->get_pointer(x, y, mask);
   }
 
--- mysql-workbench-community-6.3.4-src/library/forms/gtk/src/lf_popup.cpp	2015-06-08 08:30:54.000000000 -0300
+++ /tmp/lf_popup.cpp	2017-01-01 20:25:48.214996411 -0200
@@ -81,7 +81,8 @@
 {
   d("\n");
   Glib::RefPtr<Gdk::Colormap> colormap = screen->get_rgba_colormap();
-  _have_rgba = colormap;
+  if (colormap)
+    _have_rgba = true;
 
   if (!_have_rgba)
     colormap = screen->get_rgb_colormap();
