--- a/btuicontrols/DeviceListPage.qml	2026-05-04 16:45:16.857058438 +0200
+++ b/btuicontrols/DeviceListPage.qml	2026-05-04 23:25:41.200620481 +0200
@@ -32,12 +32,11 @@
 
     Component.onCompleted: {
         // popup.open();
+        loadDevices();
         if (permission.status === Qt.PermissionStatus.Undetermined)
             permission.request();
         else if (permission.status === Qt.PermissionStatus.Granted)
             bluetoothHandler.startDeviceDiscovery();
-
-        loadDevices();
     }
 
     Item {
@@ -116,23 +115,13 @@
         id: bluetoothHandler
 
         onNewDeviceAdded: function(name, address, uuid) {
-
-            db.executeWithParams('SELECT * FROM devices WHERE uuid = ?;', [uuid], (result) => {
-                if (result.rows.length === 0) {
-                    db.executeWithParams('INSERT INTO devices (name, address, uuid) VALUES (?, ?, ?);', [name, address, uuid]);
-                    listModel.append({name: name, address: address, isOnline: true});
-                } else {
-                    let uuid = result.rows[0].uuid;
-                    for (let i = 0; i < listModel.count; i++){
-                        if (listModel.get(i).uuid === uuid) {
-                            listModel.setProperty(i, "isOnline", true);
-                            break;
-                        }
-                    }
+            for (let i = 0; i < listModel.count; i++) {
+                if (listModel.get(i).address === address) {
+                    listModel.setProperty(i, "isOnline", true);
+                    return;
                 }
-            });
-
-            // listModel.append({name: name, address: address, isOnline: true});
+            }
+            listModel.append({name: name, address: address, isOnline: true, uuid: uuid});
         }
     }
     BluetoothPermission {
