diff -ur geigerlog.orig/geigerlog geigerlog/geigerlog
--- ./geigerlog	2019-09-02 17:47:56.000000000 +0200
+++ ./geigerlog	2019-09-03 19:22:32.694957426 +0200
@@ -6958,7 +6958,7 @@
     else:
         # dir does not exist; make it
         try:
-            os.mkdir(gglobs.dataPath )
+            os.makedirs(gglobs.dataPath)
         except:
             # dir cannot be made
             print("ERROR: main: Could not make data directory '{}'".format(gglobs.dataDirectory))
diff -ur geigerlog.orig/gutils.py geigerlog/gutils.py
--- ./gutils.py	2019-08-30 18:00:04.000000000 +0200
+++ ./gutils.py	2019-09-03 19:22:55.238956179 +0200
@@ -125,7 +125,7 @@
 
 def getDataPath():
     """Return full path of the data directory"""
-    dp = os.path.join(getProgPath(), gglobs.dataDirectory)
+    dp = os.path.join(os.getenv("HOME"), ".geigerlog")
     return dp
 
 
