diff -Naur ./geigerlog /tmp/geigerlog/geigerlog
--- ./geigerlog	2018-05-20 10:39:32.000000000 +0200
+++ /tmp/geigerlog/geigerlog	2018-05-25 11:13:09.235554108 +0200
@@ -53,6 +53,8 @@
 #   import that binding first, or set the QT_API environment variable.
 #   mplDeprecation)
 import matplotlib
+import matplotlib as mpl
+mpl.use("Qt4Agg")
 from matplotlib import __version__ as mplVersion
 
 matplotlib.use('Qt4Agg', warn=True, force=False)    # use Qt4Agg, no the default TkAgg
@@ -4847,7 +4849,7 @@
                 else:
                     # dir does not exist; make it
                     try:
-                        os.mkdir(testpath)
+                        os.makedirs(testpath)
                         gglobs.dataPath = testpath
                     except:
                         # dir cannot be made
@@ -5075,7 +5077,7 @@
     else:
         # dir does not exist; make it
         try:
-            os.mkdir(gglobs.dataPath )
+            os.makedirs(gglobs.dataPath)
         except:
             # dir cannot be made
             print("ERROR: main: Could not make data directory '{}'".format(gglobs.dataDirectory))
diff -Naur ./gutils.py /tmp/geigerlog/gutils.py
--- ./gutils.py	2018-05-19 14:52:06.000000000 +0200
+++ /tmp/geigerlog/gutils.py	2018-05-25 10:42:48.172654836 +0200
@@ -71,7 +71,7 @@
 
 def getDataPath():
     """Return full path of the data directory"""
-    dp = os.path.join(getProgPath(), gglobs.dataDirectory)
+    dp = os.path.join(os.getenv("HOME"), ".geigerlog")
     return dp
 
 
