diff -Naur ./geigerlog ../geigerlog/geigerlog
--- ./geigerlog	2021-01-25 12:18:00.000000000 +0100
+++ ../geigerlog/geigerlog	2021-01-27 23:13:10.045872597 +0100
@@ -110,7 +110,7 @@
     else:
         # dir does not exist; make it
         try:
-            os.mkdir(gglobs.dataPath )
+            os.makedirs(gglobs.dataPath )
         except:
             # dir cannot be made
             print("ERROR: main: Could not make data directory '{}'".format(gglobs.dataDirectory))
diff -Naur ./gsup_utils.py ../geigerlog/gsup_utils.py
--- ./gsup_utils.py	2021-01-26 09:44:48.000000000 +0100
+++ ../geigerlog/gsup_utils.py	2021-01-27 23:13:01.457980562 +0100
@@ -146,7 +146,7 @@
 def getDataPath():
     """Return full path of the data directory"""
 
-    dp = os.path.join(getProgPath(), gglobs.dataDirectory)
+    dp = os.path.join(os.getenv("HOME"), gglobs.dataDirectory)
     return dp
 
 
