diff -Naur /var/tmp/portage/sci-physics/geigerlog-1.0/work/geigerlog/geigerlog ./geigerlog
--- /var/tmp/portage/sci-physics/geigerlog-1.0/work/geigerlog/geigerlog	2020-10-10 12:10:20.000000000 +0200
+++ ./geigerlog	2020-10-19 23:49:39.700409001 +0200
@@ -6724,7 +6724,7 @@
     else:
         # dir does not exist; make it
         try:
-            os.mkdir(gglobs.dataPath )
+            os.makedirs(gglobs.dataPath )
         except:
             # dir cannot be made
             print("ERROR: main: Could not make data directory '{}'".format(gglobs.dataDirectory))
diff -Naur /var/tmp/portage/sci-physics/geigerlog-1.0/work/geigerlog/gutils.py ./gutils.py
--- /var/tmp/portage/sci-physics/geigerlog-1.0/work/geigerlog/gutils.py	2020-10-10 11:24:36.000000000 +0200
+++ ./gutils.py	2020-10-19 23:49:58.319408052 +0200
@@ -131,7 +131,7 @@
 
 def getDataPath():
     """Return full path of the data directory"""
-    dp = os.path.join(getProgPath(), gglobs.dataDirectory)
+    dp = os.path.join(os.getenv("HOME"), gglobs.dataDirectory)
     return dp
 
 
